### PR TITLE
[REVIEW] Handle behavior when the optimized plan contains a LogicalValues.

### LINKED ISSUE #429: [BUG] engine crashes if you write a sql clause where the between is not in ascending order
**Describe the bug**
between 0 and 5 works
between 5 and 0 crashes the engine

### PR DESCRIPTION
**Pull request by Christian8491**
Closes #429
Simple case:

`query = "select n_nationkey FROM nation WHERE n_nationkey BETWEEN 5 AND 1"`

Then the logical plans are:

```
DEBUG: com.blazingdb.calcite.application.RelationalAlgebraGenerator - optimized
LogicalValues(tuples=[[]])

DEBUG: com.blazingdb.calcite.application.RelationalAlgebraGenerator - non optimized
LogicalProject(n_nationkey=[$0])
  LogicalFilter(condition=[AND(>=($0, 5), <=($0, 1))])
    LogicalTableScan(table=[[main, nation]])
```

### CODE REVIEW

**Comment by felipeblazing** [comment]
This returns this because the query "select n_nationkey FROM nation WHERE n_nationkey BETWEEN 5 AND 1" will always just be an empty set, there is no number that is greater than 5 and less t han 1 in the space of integers :)

**Comment by Christian8491** [comment]
Yep.

**Comment by williamBlazing** [comment]
@Christian8491 El motor deberia de dar un error para cualquier nodo de algebra relacional no reconocido. Tu fix solo arregla un sintoma del problema. El problema es que el motor crashea si le pasas un paso de algebra relacional no conosido. Queremos que si llega otra cosa no reconosida, que el motor le pase un error al usuario mas util, como `Error: Relational algebra node LogicalValues not implemented` y LogicalValues deberia ser lo que sea. Por ejemplo si tratas de hacer un Windowed function, deberia darte un error similar

**Comment by felipeblazing** [comment]
To be clear LogicalValues should be valid at some point we just happen not
to reocgnize it now. Actually the very best thing to result from this query
is to return an empty set, though the user should realize it will always be
empty

**Comment by kharoc** [comment]
1 RAL

========================================================
FAILED TESTS                                                                                                                                                                                                
========================================================                                                                                                                                                    
                        TestGroup InputType Result                                           TestId
0  Bindable Alias                  orc       Fail   TEST_09                                        
1  Date                            orc       Fail   TEST_01                                        
2  Simple Distribution From Local  orc       Fail   TEST_22,TEST_43,TEST_46,TEST_49                
3  TPCH Queries                    orc       Fail   TEST_03,TEST_05,TEST-08,TEST_10,TEST_14,TEST_18
4  Timestamp                       orc       Fail   TEST_02                                        
*******************************************************************************
TOTAL SUMMARY for test suite: 
PASSED: 1261/1274
FAILED: 13/1274
CRASH: 0/1274
TOTAL: 1274
MAX DELTA: 48.0
*******************************************************************************
Aggregations without group by:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Coalesce:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Column Basis:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Common Table Expressions:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Count without group by:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Date:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Timestamp:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Full outer join:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Group by:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Group by without aggregations:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Inner join:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Left outer join:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Non-EquiJoin Queries:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Nested Queries:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Order by:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Predicates With Nulls:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Simple String:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Unary ops:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Unify Tables:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Union:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Limit:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Where clause:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Bindable Alias:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Boolean:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Case:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Cast:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Concat:   Start Mem: 8117.875   End Mem: 8165.875   Diff: 48.0
Literal:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
Dir:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
File System Local:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
File System S3:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
File System Google Storage :   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
Like:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
Substring:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
Wild Card:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
TPCH Queries:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
Round:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0

2 Worker's

========================================================
FAILED TESTS
========================================================
                        TestGroup InputType Result                                           TestId
0  Bindable Alias                  orc       Fail   TEST_09                                        
1  Date                            orc       Fail   TEST_01                                        
2  Simple Distribution From Local  orc       Fail   TEST_22,TEST_43,TEST_46,TEST_49                
3  TPCH Queries                    orc       Fail   TEST_03,TEST_05,TEST-08,TEST_10,TEST_14,TEST_18
4  Timestamp                       orc       Fail   TEST_02                                        
*******************************************************************************
TOTAL SUMMARY for test suite: 
PASSED: 1500/1513
FAILED: 13/1513
CRASH: 0/1513
TOTAL: 1513
MAX DELTA: 457.0
*******************************************************************************
Aggregations without group by:   Start Mem: 864.875   End Mem: 1321.875   Diff: 457.0
Coalesce:   Start Mem: 1321.875   End Mem: 1405.875   Diff: 84.0
Common Table Expressions:   Start Mem: 1405.875   End Mem: 1413.875   Diff: 8.0
Count without group by:   Start Mem: 1413.875   End Mem: 1427.875   Diff: 14.0
Date:   Start Mem: 1427.875   End Mem: 1505.875   Diff: 78.0
Timestamp:   Start Mem: 1505.875   End Mem: 1551.875   Diff: 46.0
Full outer join:   Start Mem: 1551.875   End Mem: 1593.875   Diff: 42.0
Group by:   Start Mem: 1593.875   End Mem: 1613.875   Diff: 20.0
Group by without aggregations:   Start Mem: 1613.875   End Mem: 1617.875   Diff: 4.0
Inner join:   Start Mem: 1617.875   End Mem: 1655.875   Diff: 38.0
Left outer join:   Start Mem: 1655.875   End Mem: 1675.875   Diff: 20.0
Non-EquiJoin Queries:   Start Mem: 1675.875   End Mem: 1779.875   Diff: 104.0
Nested Queries:   Start Mem: 1779.875   End Mem: 1813.875   Diff: 34.0
Order by:   Start Mem: 1813.875   End Mem: 1839.875   Diff: 26.0
Predicates With Nulls:   Start Mem: 1839.875   End Mem: 1891.875   Diff: 52.0
Simple String:   Start Mem: 1891.875   End Mem: 1957.875   Diff: 66.0
Unary ops:   Start Mem: 2277.875   End Mem: 2331.875   Diff: 54.0
Unify Tables:   Start Mem: 2331.875   End Mem: 2411.875   Diff: 80.0
Union:   Start Mem: 2411.875   End Mem: 2393.875   Diff: -18.0
Limit:   Start Mem: 2393.875   End Mem: 2269.875   Diff: -124.0
Where clause:   Start Mem: 2269.875   End Mem: 2359.875   Diff: 90.0
Bindable Alias:   Start Mem: 2359.875   End Mem: 2501.875   Diff: 142.0
Boolean:   Start Mem: 2501.875   End Mem: 2541.875   Diff: 40.0
Case:   Start Mem: 2541.875   End Mem: 2671.875   Diff: 130.0
Cast:   Start Mem: 2671.875   End Mem: 2757.875   Diff: 86.0
Concat:   Start Mem: 2757.875   End Mem: 2963.875   Diff: 206.0
Literal:   Start Mem: 2963.875   End Mem: 2975.875   Diff: 12.0
Dir:   Start Mem: 2975.875   End Mem: 3345.875   Diff: 370.0
File System Local:   Start Mem: 3345.875   End Mem: 3373.875   Diff: 28.0
File System S3:   Start Mem: 3373.875   End Mem: 3551.875   Diff: 178.0
File System Google Storage :   Start Mem: 3551.875   End Mem: 3727.875   Diff: 176.0
Like:   Start Mem: 3727.875   End Mem: 3997.875   Diff: 270.0
Substring:   Start Mem: 4917.875   End Mem: 5053.875   Diff: 136.0
Wild Card:   Start Mem: 5053.875   End Mem: 5471.875   Diff: 418.0
TPCH Queries:   Start Mem: 5471.875   End Mem: 5835.875   Diff: 364.0
Round:   Start Mem: 5835.875   End Mem: 5935.875   Diff: 100.0

File: engine/src/execution_graph/logic_controllers/PhysicalPlanGenerator.h [hunk 1]
```diff
@@ -108,6 +108,8 @@ struct tree_processor {
 			k = std::make_shared<UnionKernel>(expr, kernel_context, query_graph);
 			kernel_context->setKernelId(k->get_id());
 			k->set_type_id(kernel_type::UnionKernel);
+		} else {
```
Comment (felipeblazing) [inline engine/src/execution_graph/logic_controllers/PhysicalPlanGenerator.h:111]: can you make a custom exception that can include the snippet of algebra that coudl not be interpreted and return that exception so we can catch this sepcific problem
that way we can catch BlazingAlgebraNotSupportedException and tell the user what part of the algebra plan was not supported

File: pyblazing/pyblazing/apiv2/context.py [hunk 2]
```diff
@@ -1599,6 +1600,11 @@ def sql(self, sql, table_list=[], algebra=None, return_futures=False, single_gpu
         if (algebra is None):
             algebra = self.explain(sql)
 
+        # when `LogicalValues` appears on the optimized plan there aren't neither BindableTableScan nor TableScan nor Project
+        if "LogicalValues(tuples=[[]])" in algebra:
+            print("This SQL statement returns empty result. Please double check your query.")
+            return
```
Comment (felipeblazing) [inline pyblazing/pyblazing/apiv2/context.py:1606]: it should return an empty dataframe this return None which is wrong

Comment (Christian8491) [reply pyblazing/pyblazing/apiv2/context.py:1606]: the algebra at this point does not contain neither BindableTableScan, neither TableScan neither Projects. Then the empty dataframe doesn't have  a column name. Should I use  a  col_name as  'EMPTY' or something like that?

File: engine/src/execution_graph/logic_controllers/PhysicalPlanGenerator.h (hunk 1 above)
Comment (Christian8491) [reply engine/src/execution_graph/logic_controllers/PhysicalPlanGenerator.h:111]: Something like this
`throw std::runtime_error("expression in the Algebra Relational is currently not supported: " + expr);`
OR do you thing another type of exception?

File: pyblazing/pyblazing/apiv2/context.py (hunk 2 above)
Comment (felipeblazing) [reply pyblazing/pyblazing/apiv2/context.py:1606]: cant you just return cudf.DataFrame()

Comment (felipeblazing) [reply pyblazing/pyblazing/apiv2/context.py:1606]: import cudf
cudf.DataFrame()

works for me

File: engine/src/execution_graph/logic_controllers/PhysicalPlanGenerator.h (hunk 1 above)
Comment (felipeblazing) [reply engine/src/execution_graph/logic_controllers/PhysicalPlanGenerator.h:111]: like 

```
throw BlazingSqlInvalidAlgebraException("info here")
```

Comment (felipeblazing) [reply engine/src/execution_graph/logic_controllers/PhysicalPlanGenerator.h:111]: dont throw std::runtime_error that can be anything

Comment (Christian8491) [reply engine/src/execution_graph/logic_controllers/PhysicalPlanGenerator.h:112]: changes were made

**Comment by kharoc** [comment]
1 RAL

========================================================
FAILED TESTS                                                                                                                                                                                                
========================================================                                                                                                                                                    
                        TestGroup InputType Result                                           TestId
0  Bindable Alias                  orc       Fail   TEST_09                                        
1  Date                            orc       Fail   TEST_01                                        
2  Simple Distribution From Local  orc       Fail   TEST_22,TEST_43,TEST_46,TEST_49                
3  TPCH Queries                    orc       Fail   TEST_03,TEST_05,TEST-08,TEST_10,TEST_14,TEST_18
4  Timestamp                       orc       Fail   TEST_02                                        
*******************************************************************************
TOTAL SUMMARY for test suite: 
PASSED: 1261/1274
FAILED: 13/1274
CRASH: 0/1274
TOTAL: 1274
MAX DELTA: 48.0
*******************************************************************************
Aggregations without group by:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Coalesce:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Column Basis:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Common Table Expressions:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Count without group by:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Date:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Timestamp:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Full outer join:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Group by:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Group by without aggregations:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Inner join:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Left outer join:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Non-EquiJoin Queries:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Nested Queries:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Order by:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Predicates With Nulls:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Simple String:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Unary ops:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Unify Tables:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Union:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Limit:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Where clause:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Bindable Alias:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Boolean:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Case:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Cast:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Concat:   Start Mem: 8117.875   End Mem: 8165.875   Diff: 48.0
Literal:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
Dir:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
File System Local:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
File System S3:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
File System Google Storage :   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
Like:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
Substring:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
Wild Card:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
TPCH Queries:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
Round:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0

2 Worker's

<========================================================
FAILED TESTS                                                                                                                                                                                                
========================================================                                                                                                                                                    
                        TestGroup InputType Result                                           TestId
0  Bindable Alias                  orc       Fail   TEST_09                                        
1  Date                            orc       Fail   TEST_01                                        
2  Simple Distribution From Local  orc       Fail   TEST_22,TEST_43,TEST_46,TEST_49                
3  TPCH Queries                    orc       Fail   TEST_03,TEST_05,TEST-08,TEST_10,TEST_14,TEST_18
4  Timestamp                       orc       Fail   TEST_02                                        
*******************************************************************************
TOTAL SUMMARY for test suite: 
PASSED: 1261/1274
FAILED: 13/1274
CRASH: 0/1274
TOTAL: 1274
MAX DELTA: 48.0
*******************************************************************************
Aggregations without group by:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Coalesce:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Column Basis:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Common Table Expressions:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Count without group by:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Date:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Timestamp:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Full outer join:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Group by:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Group by without aggregations:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Inner join:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Left outer join:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Non-EquiJoin Queries:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Nested Queries:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Order by:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Predicates With Nulls:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Simple String:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Unary ops:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Unify Tables:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Union:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Limit:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Where clause:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Bindable Alias:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Boolean:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Case:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Cast:   Start Mem: 8117.875   End Mem: 8117.875   Diff: 0.0
Concat:   Start Mem: 8117.875   End Mem: 8165.875   Diff: 48.0
Literal:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
Dir:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
File System Local:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
File System S3:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
File System Google Storage :   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
Like:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
Substring:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
Wild Card:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
TPCH Queries:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0
Round:   Start Mem: 8165.875   End Mem: 8165.875   Diff: 0.0

**Pull request merged**